app/lib/features/invite_members/widgets/space_member_invite_card.dart

```diff
@@ -48,8 +48,24 @@ class SpaceMemberInviteCard extends ConsumerWidget {
     final roomId = space.getRoomIdStr();
     final title =
         profile.displayName?.isNotEmpty == true ? profile.displayName! : roomId;
-    final parent = ref.watch(canonicalParentProvider(roomId)).valueOrNull;
-    final parentRoomId = parent?.space.getRoomIdStr();
+    final canonicalParents = ref.watch(canonicalParentsProvider(roomId));
+    List<AvatarInfo> parentBadges = List.empty(growable: true);
+
+    if (canonicalParents.valueOrNull != null) {
+      final parents = canonicalParents.requireValue;
+      if (parents.isNotEmpty) {
+        parentBadges = parents.map((e) {
+          final roomId = e.space.getRoomIdStr();
+          final displayName = e.profile.displayName ?? roomId;
+          final avatar = e.profile.getAvatarImage();
+          return AvatarInfo(
+            uniqueId: roomId,
+            displayName: displayName,
+            avatar: avatar,
+          );
+        }).toList();
+      }
```

Comment:
so ... this happens so often, it screams like that should be its own provider, don't you think? then we could just have a single `parentBadges: ref.watch(parentAvatarInfos(roomId)),` which is a quite a bit nicer and

Reply:
Yeah I think there's a room for simplification here! 👍🏻